### PR TITLE
Add doc/engines.md

### PR DESCRIPTION
**Pull request by dj2**
This Cl adds documentation on the engine layer. This includes
information on the engine lifecycle along with brief notes on each
of the methods in the engine interface.

### CODE REVIEW

**Comment by dj2** [comment]
First pass brain dump.... What did I miss?

File: docs/engines.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+depth attachments for the given pipeline.
+
+
+#### `DoDrawRect`
+The `DoDrawRect` instructs the engine to draw the given pipeline in the box
+at (x,y) of size (width,height). The buffers must be read back into the backing
+`amber::Buffer` at the end of this method.
+
+
+### `DoDrawArrays`
```
Comment (dneto0) [inline docs/engines.md]: This one is at a higher level of header. (3 rather than 4)

File: docs/engines.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+The `DoPatchParameterVertices` tells the engine to do a thing with some
+values....I don't really know what it means, heh.
+
+
+#### `DoBuffer`
+The `DoBuffer` command tells the engine that for a given pipeline the given
+data should be updated into the given buffer.
+
+
+### `SetEngineData`
```
Comment (dneto0) [inline docs/engines.md]: This one is at a higher level of header. (3 rather than 4)

File: docs/engines.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+the provided (x,y,z) parameters.
+
+
+#### `DoEntryPoint`
+The `DoEntryPoint` command instructs the engine to change the entry point in
+the given pipeline for the give shader.
+
+
+#### `DoPatchParameterVertices`
+The `DoPatchParameterVertices` tells the engine to do a thing with some
```
Comment (dneto0) [inline docs/engines.md]: Lol.  Me neither.

File: docs/engines.md (hunk 1 above)
Comment (dj2) [reply docs/engines.md]: done

File: docs/engines.md (hunk 2 above)
Comment (dj2) [reply docs/engines.md]: done

**Pull request merged**